src/components/ui/Chat/ChannelList/ChannelItem.tsx

```diff
@@ -22,25 +22,79 @@ export interface ChannelItemProps
   onClick: () => void;
-  /** Signifies of there are unread message in the channel. */
+  /** Signifies if there are unread message in the channel. */
   unread?: boolean;
   /** Content of the badge signaling unread messages, such as a badge displaying the number. */
   unreadBadgeLabel?: string;
+  /* Content of the last channel message to be displayed */
+  lastChannelMessage?: string;
+  /* Timestamp of the last channel message to be displayed */
+  lastChannelMessageTimestamp?: string;
 }
 
 export const ChannelItem: FC<ChannelItemProps> = (props) => {
-  const { name, actions, isSelected, onClick, unread, unreadBadgeLabel } =
-    props;
+  const {
+    name,
+    actions,
+    isSelected,
+    onClick,
+    unread,
+    unreadBadgeLabel,
+    lastChannelMessage,
+    lastChannelMessageTimestamp,
+  } = props;
+
+  const displayDetailedChannelView =
+    lastChannelMessage || lastChannelMessageTimestamp;
+  const displayUnreadBadgeForDetailedView =
+    displayDetailedChannelView && unread && unreadBadgeLabel;
+  const displayUnreadBadgeForSimpleView =
+    !displayDetailedChannelView && unread && unreadBadgeLabel;
+  const displayPopOverForDetailedView =
+    displayDetailedChannelView && actions && isSelected;
+  const displayPopOverForSimpleView =
+    !displayDetailedChannelView && actions && isSelected;
   return (
     <StyledChannelItem
       {...props}
       className={classnames({ 'ch-selected': isSelected, 'ch-unread': unread })}
     >
-      <Button className="ch-channel-button" label={name} onClick={onClick} />
-      {unread && unreadBadgeLabel && (
-        <Badge value={unreadBadgeLabel} className="ch-unread-badge" />
+      {displayDetailedChannelView ? (
+        <div className={'ch-detailed-channel'} onClick={onClick}>
+          <div className="ch-detailed-channel-name">{name}</div>
+          <div className="ch-detailed-channel-message">
+            {lastChannelMessage}
+          </div>
+          <div className="ch-detailed-channel-message-time">
+            {lastChannelMessageTimestamp}
+          </div>
+        </div>
+      ) : (
+        <Button className="ch-channel-button" label={name} onClick={onClick} />
+      )}
+      {displayUnreadBadgeForDetailedView && (
+        <Badge value={unreadBadgeLabel!} className="ch-unread-badge-detailed" />
+      )}
+      {displayUnreadBadgeForSimpleView && (
+        <Badge value={unreadBadgeLabel!} className="ch-unread-badge" />
+      )}
+      {displayPopOverForDetailedView && (
+        <PopOver
+          className={'ch-popover-toggle-detailed'}
+          a11yLabel="Open channel options"
+          placement="bottom-end"
+          renderButton={(isOpen: boolean) => (
+            <Dots
+              width="1.5rem"
+              height="1.5rem"
+              className={`${isOpen ? 'isOpen' : ''} ch-channel-actions`}
+              data-testid="channel-actions"
+            />
+          )}
+          children={actions}
+        />
```

Comment:
I'd prefer if we could group the DetailedView and the SimpleView together, instead of doing this "back-and-forth" style. 

Maybe it could be simpler if you did like:
```
detailedView && (
    <channel component>
    {unread && unreadBadgeLabel && <Badge value={unreadBadgeLabel!} className="ch-unread-badge-detailed" />}
    {displayPopOverForDetailedView && (
        <PopOver
          className={'ch-popover-toggle-detailed'}
          a11yLabel="Open channel options"
          placement="bottom-end"
          renderButton={(isOpen: boolean) => (
            <Dots
              width="1.5rem"
              height="1.5rem"
              className={`${isOpen ? 'isOpen' : ''} ch-channel-actions`}
              data-testid="channel-actions"
            />
          )}
          children={actions}
        />
     )}
)

simpleView && (
    <Button className="ch-channel-button" label={name} onClick={onClick} />
    {unread && unreadBadgeLabel && <Badge value={unreadBadgeLabel!} className="ch-unread-badge" />}
    {displayPopOverForSimpleView && (
            <PopOver
          a11yLabel="Open channel options"
          placement="bottom-end"
          renderButton={(isOpen: boolean) => (
            <Dots
              width="1.5rem"
              height="1.5rem"
              className={`${isOpen ? 'isOpen' : ''} ch-channel-actions`}
              data-testid="channel-actions"
            />
          )}
          children={actions}
        />
    )}
)
```

Reply:
maybe you can create just one const for `unread && unreadBadgeLabel`

Reply:
Thanks! Addressed your comment.